source/ref_count.c

```diff
@@ -73,10 +61,18 @@ static bool s_thread_count_zero_pred(void *user_data) {
     return s_global_thread_count == 0;
 }
 
-void aws_global_thread_shutdown_wait(void) {
-    aws_common_fatal_assert_library_initialized();
-
+void aws_global_thread_creator_shutdown_wait(void) {
     aws_mutex_lock(&s_global_thread_lock);
     aws_condition_variable_wait_pred(&s_global_thread_signal, &s_global_thread_lock, s_thread_count_zero_pred, NULL);
     aws_mutex_unlock(&s_global_thread_lock);
 }
+
+void aws_global_thread_creator_shutdown_wait_for(uint32_t wait_timeout_in_seconds) {
+    int64_t wait_time_in_nanos =
+        aws_timestamp_convert(wait_timeout_in_seconds, AWS_TIMESTAMP_SECS, AWS_TIMESTAMP_NANOS, NULL);
+
+    aws_mutex_lock(&s_global_thread_lock);
+    aws_condition_variable_wait_for_pred(
+        &s_global_thread_signal, &s_global_thread_lock, wait_time_in_nanos, s_thread_count_zero_pred, NULL);
+    aws_mutex_unlock(&s_global_thread_lock);
```

Comment:
The function declaration says "Crashes via fatal assert if the wait timeout is reached." but we're not actually checking whether the condition variable timed out.

I'd actually prefer we raise an error, vs doing a fatal assert, since we'll be calling this from tests in other languages and can turn an error into an idiomatic Exception, vs a scary crash in C

Reply:
Ok, will return success/error and wrap the calls in ASSERT_SUCCESS